lib/puppet/application/lookup.rb

```diff
@@ -57,6 +57,102 @@ class Puppet::Application::Lookup < Puppet::Application
     end
   end
 
+  def help
+    <<-'HELP'
+
+puppet-lookup(8) -- Data in modules lookup function
+========
+
+SYNOPSIS
+--------
+The lookup command is used for debugging and testing a given data
+configuration. For a given data key, lookup will return which value
+will be produced with an optional explanation of how the system arrived
+at that value.
```

Comment:
Both value and explanation will not be returned in the the implementation that is pending for PUP-4475. It's either or. Reason being that with plain text explain, the result is clearly visible within the explanation, and if 'explain_json' (new option) is used, the producing the value in addition to the json output would clobber the value. If we decide that that's the way to go, then this doc needs to be updated.